app/test/features/chat_ng/widgets/chat_messages_list_test.dart

```diff
@@ -0,0 +1,406 @@
+        find.byType(AnimatedList),
+      );
+      expect(animatedList.initialItemCount, equals(5));
+    });
+
+    testWidgets('shows loading indicator when loading', (tester) async {
+      final loadingState = ChatRoomState(
+        messageList: ['msg1', 'msg2'],
+        loading: const ChatRoomLoadingState.loading(),
+      );
+
+      await tester.pumpProviderWidget(
+        overrides: [
+          chatMessagesStateProvider(
+            testRoomId,
+          ).overrideWith((ref) => MockChatRoomMessagesNotifier(loadingState)),
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => GlobalKey<AnimatedListState>()),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      expect(find.byType(CircularProgressIndicator), findsOneWidget);
+    });
+
+    testWidgets('hides loading indicator when not loading', (tester) async {
+      final loadedState = ChatRoomState(
+        messageList: ['msg1', 'msg2'],
+        loading: const ChatRoomLoadingState.loaded(),
+      );
+
+      await tester.pumpProviderWidget(
+        overrides: [
+          chatMessagesStateProvider(
+            testRoomId,
+          ).overrideWith((ref) => MockChatRoomMessagesNotifier(loadedState)),
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => GlobalKey<AnimatedListState>()),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      expect(find.byType(CircularProgressIndicator), findsNothing);
+    });
+
+    testWidgets('automatically scrolls to bottom on initial load', (
+      tester,
+    ) async {
+      final initialState = ChatRoomState(
+        messageList: [],
+        loading: const ChatRoomLoadingState.loaded(),
+      );
+
+      final notifier = ChatRoomMessagesNotifier(
+        roomId: testRoomId,
+        ref: MockRef(),
+      );
+      notifier.state = initialState;
+
+      final animatedListKey = GlobalKey<AnimatedListState>();
+
+      await tester.pumpProviderWidget(
+        overrides: [
+          chatMessagesStateProvider(testRoomId).overrideWith((ref) => notifier),
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => animatedListKey),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      // Update with messages to trigger auto-scroll
+      final updatedState = initialState.copyWith(
+        messageList: List.generate(5, (index) => 'msg$index'),
+      );
+      notifier.state = updatedState;
```

Comment:
does this actually work? I thought it was necessary to do actually modify the notifier via the `ref` and not just standalone. And after it is not tested in here that the change actually did anything on the object, no check whether more items are rendered. I suspect the test passes because this change doesn't actually happen. Can we have additional checks that the update actually took place.

Reply:
Yes, I've modified the test case to do actual update on notifier and added additional checks ! 👍🏻